File: src/Lessons/5-PrefixSums/5.1-PassingCars.js
function solution(A) {
    //Find the number of pairs of passing cars
    let array = [...A];
    let Pcars = 0; //if 0

    let pairs = 0;
    for (let i = 0; i < array.length; i++) {
        if (array[i] == 0)
            Pcars++;
        else
            pairs += Pcars;
            
        if (pairs > 1000000000) {
            pairs = -1;
            break;
        }
        //console.log('pcars ' +Pcars + ' ::: ' +pairs);
    }

    return pairs;
}

const testArray = [0, 1, 0, 1, 1];
console.log(solution(testArray))
//expect 7

File: src/Lessons/5-PrefixSums/5.1-PassingCars.go
package main

import "fmt"

func main() {
	fmt.Println(solution([]int{0, 1, 0, 1, 1}))
}
func solution(someArray []int) int {
	passingCars := 0
	pairs := 0

	for index := 0; index < len(someArray); index++ {
		if someArray[index] == 0 {
			passingCars++
		} else {
			pairs += passingCars
		}
		if pairs > 1000000000 {
			pairs = -1
			break
		}
		//fmt.Println("Car:: ", someArray[index])
		//fmt.Println("pcars:: ", passingCars, " pairs:: ", pairs)
	}
	return pairs
}
